### DISCUSSION TITLE
Uno.UI.SourceGenerators generated files not existing on disk?

### DISCUSSION THREAD

**Discussion by sigmarsson** [unoplatform/uno#6509] [Q&A] [answered]
## Current behavior

There is a compiler error I am supposed to fix but this gen file does not exist on disk. how can see the error's context ?

`..\Uno.UI.SourceGenerators\Uno.UI.SourceGenerators.XamlGenerator.XamlCodeGenerator\App_d06e9e3eec7aad7fbddeb7894127a713.cs`

## Environment

Nuget Package:
```
  <ItemGroup>
    <PackageReference Include="Microsoft.Extensions.Logging" Version="6.0.0-preview.6.21352.12" />
    <PackageReference Include="Microsoft.Windows.Compatibility" Version="6.0.0-preview.6.21352.12" />
    <PackageReference Include="Prism.DryIoc.Extensions" Version="8.0.62" />
    <PackageReference Include="Prism.DryIoc.Uno.WinUI" Version="8.1.97" />
    <PackageReference Include="Prism.Uno.WinUI" Version="8.1.97" />
    <PackageReference Include="Uno.Extensions.Logging.WebAssembly.Console" Version="1.2.0-dev.2" />
    <PackageReference Include="Uno.WinUI.WebAssembly" Version="3.9.0-dev.337" />
    <PackageReference Include="Uno.WinUI.RemoteControl" Version="3.9.0-dev.337" Condition="'$(Configuration)'=='Debug'" />
    <PackageReference Include="Uno.Wasm.Bootstrap" Version="3.0.0-dev.101" />
    <PackageReference Include="Uno.Wasm.Bootstrap.DevServer" Version="3.0.0-dev.101" />
```
  </ItemGroup>

- [ ] Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia
- [x] Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia
- [x] Uno.SourceGenerationTasks
- [ ] Uno.UI.RemoteControl / Uno.WinUI.RemoteControl
- [ ] Other: 

Nuget Package Version(s):

Affected platform(s):

- [ ] iOS
- [ ] Android
- [x] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

IDE:

- [x] Visual Studio 2022 (version: 17.0.0 P2)
- [ ] Visual Studio 2017 (version: )
- [ ] Visual Studio 2019 (version: )
- [ ] Visual Studio for Mac (version: )
- [ ] Rider Windows (version: )
- [ ] Rider macOS (version: )
- [ ] Visual Studio Code (version: )

Relevant plugins:

- [ ] Resharper (version: )

**Comment by jeromelaban** [comment]
Thanks for the report. When building for WebAssembly, generated files are kept in memory by the Roslyn compiler. You can view those in Visual Studio in the `Analyzers` node of the project.

**Reply by sigmarsson** [reply]
Thank you @jeromelaban !

I am trying to follow through what you have just said but the respective generator has not got child nodes. Please see sshot.  `Uno.UI.SourceGenerators`

![image](https://user-images.githubusercontent.com/43340959/126484650-0d0a801b-8aab-4a65-a5a5-a64a61fc02bb.png)

**Reply by jeromelaban** [reply]
This feels like a Visual Studio issue. You can also restore the previous source generators with this:

```xml
<PropertyGroup>
	<UnoUIUseRoslynSourceGenerators>false</UnoUIUseRoslynSourceGenerators>
</ProperyGroup>
```

**Reply by sigmarsson** [reply]
In that case the error is : `Identifier Expected` for all event handlers.

![image](https://user-images.githubusercontent.com/43340959/129930023-7a18490c-052c-412c-89e4-05b2ce137dd1.png)

**Reply by sigmarsson** [reply]
![image](https://user-images.githubusercontent.com/43340959/129960162-b22e892e-89e5-4fd6-a864-0e98c98d4f7e.png)

**Comment by Youssef1313** [comment, marked as answer]
You can add the following in a `PropertyGroup` in a `csproj`:

```xml
    <EmitCompilerGeneratedFiles>true</EmitCompilerGeneratedFiles>
    <CompilerGeneratedFilesOutputPath>Generated</CompilerGeneratedFilesOutputPath>
```

**NOTE:** For some reason the build will fail when the generated files already exists on disk. So you'll have to delete them before rebuilding (just use this as a temporary solution to see the generated files easily).

**Comment by sigmarsson** [comment]
Thanks, it seems to has taken affect. Why is it happening ? ExportView_4d19ee40cc650e2e5c1d319b9195a69e.cs(202,65,202,66): error CS1001: Identifier expected



<img width="775" alt="image" src="https://github.com/unoplatform/uno/assets/43340959/78b0fd0e-308c-465d-87c5-75af12e53f61">


```
.GenericApply(__that, __nameScope, ((c5, __that, __nameScope) => 
{
				c5.SetBinding(
					global::Microsoft.UI.Xaml.Controls.CheckBox.ContentProperty,
					new Microsoft.UI.Xaml.Data.Binding()
				);
				var Checked_ViewModel_OnSelect_That = (__ResourceOwner_1 as global::Uno.UI.DataBinding.IWeakReferenceProvider).WeakReference;
				/* second level */ c5.Checked += (ViewModel.OnSelect_sender,ViewModel.OnSelect_e) => (Checked_ViewModel_OnSelect_That.Target as global::Weather.History.Mvvm.Views.ExportView)?.ViewModel.OnSelect(ViewModel.OnSelect_sender,ViewModel.OnSelect_e);
				var Unchecked_ViewModel_OnDeselect_That = (__ResourceOwner_1 as global::Uno.UI.DataBinding.IWeakReferenceProvider).WeakReference;
				/* second level */ c5.Unchecked += (ViewModel.OnDeselect_sender,ViewModel.OnDeselect_e) => (Unchecked_ViewModel_OnDeselect_That.Target as global::Weather.History.Mvvm.Views.ExportView)?.ViewModel.OnDeselect(ViewModel.OnDeselect_sender,ViewModel.OnDeselect_e);
				global::Uno.UI.FrameworkElementHelper.SetBaseUri(c5, __baseUri_ExportView_4d19ee40cc650e2e5c1d319b9195a69e, "file:///C:/Users/weath/source/repos/Weather.History/uno/Weather.History.Shared/Mvvm/Views/ExportView.xaml", 27, 22);
				c5.CreationComplete();
}
))
```

**Reply by jeromelaban** [reply]
This looks like a bug. Can you open an issue with a repro? Thanks!
